pkg/code-generator/codegen/templates/resource_template.go

```diff
@@ -87,8 +87,67 @@ func (r *{{ .Name }}) SetMetadata(meta *core.Metadata) {
 {{- if $.HasStatus }}
 
 func (r *{{ .Name }}) SetStatus(status *core.Status) {
-	r.Status = status
+	r.AddToReporterStatus(status)
 }
+
+func (r *{{ .Name }}) SetReporterStatus(status *core.ReporterStatus) {
+	r.StatusOneof = &{{ .Name }}_ReporterStatus{ReporterStatus: status}
+}
+
+// AddToReporterStatus inserts the specified status into the ReporterStatus.Statuses map for the
+// controller specified by status.ReportedBy.  If the resource does not yet have a ReporterStatus,
+// one will be created.
+// Note: POD_NAMESPACE environment variable must be set for this function to behave as expected.
+func (r *{{ .Name }}) AddToReporterStatus(status *core.Status) {
+	podNamespace := os.Getenv("POD_NAMESPACE")
+	if podNamespace != "" {
+		if r.GetReporterStatus() == nil {
+			r.SetReporterStatus(&core.ReporterStatus{})
+		}
+		if r.GetReporterStatus().Statuses == nil {
+			r.GetReporterStatus().Statuses = make(map[string]*core.Status)
+		}
+		key := podNamespace + ":" + status.GetReportedBy()
+		r.GetReporterStatus().Statuses[key] = status
+	}
+}
+
+// GetStatusForReporter returns the status stored in the ReporterStatus.Statuses map for the
+// controller specified by reportedBy, or nil if no status exists for that controller.
+// Note: POD_NAMESPACE environment variable must be set for this function to behave as expected.
+func (r *{{ .Name }}) GetStatusForReporter(reportedBy string) *core.Status {
+	podNamespace := os.Getenv("POD_NAMESPACE")
+	if podNamespace != "" {
+		key := podNamespace + ":" + reportedBy
```

Comment:
the old `GetStatus()` didn't require `reportedBy` to be provided by the user. we can probably simplify this to avoid passing in the arg here since `reportedBy` today will always be the same controller for gloo

Reply:
The `reportedBy` field is needed for the key in the `ReporterStatus.Statuses` map.  Or are you suggesting we change that map key from "POD_NAMESPACE:reportedBy" to just "POD_NAMESPACE"?

Reply:
I'm just noting that the original function didn't have any params and it still said "reported_by: gloo" or "reported_by: gateway" and that we should be able to lean on that

Reply:
There is still the `GetReporterStatus()` variation which does not have any params, but the caller would then need to locate the `Status` in the map that's relevant to them.  The idea behind `GetStatusForReporter(reportedBy string)` is to be a helper function so that each caller doesn't need to know the map key format, and can instead get the `Status` they care about directly.